From c4c84589f50cddabdb894d4795c73ac74b6f70d3 Mon Sep 17 00:00:00 2001
From: ghudson <ghudson@dc483132-0cff-0310-8789-dd5450dbe970>
Date: Thu, 22 Mar 2012 14:07:44 +0000
Subject: =?UTF-8?q?Suppress=20some=20gcc=20uninitialized=20variable=20warnin?=
 =?UTF-8?q?gs=0Aticket:=207107=0Agcc=204.6.2=20reportedly=20finds=20some=20s?=
 =?UTF-8?q?purious=20maybe-uninitialized=20warnings.=0ASuppress=20them.=20=20?=
 =?UTF-8?q?Patch=20from=20Eray=20Aslan=20with=20some=20adjustment.?=

git-svn-id: svn://anonsvn.mit.edu/krb5/trunk@25783 dc483132-0cff-0310-8789-dd5450dbe970
(cherry picked from commit 254c31d2be4e5250932e7701c68d23a27edec8de)

Patch-Category: upstream
---
 src/kdc/do_as_req.c           |    1 +
 src/lib/kadm5/srv/svr_iters.c |    2 +-
 src/slave/kprop.c             |    5 +++--
 3 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/src/kdc/do_as_req.c b/src/kdc/do_as_req.c
index 23623fe..79bb645 100644
--- a/src/kdc/do_as_req.c
+++ b/src/kdc/do_as_req.c
@@ -191,6 +191,7 @@ finish_process_as_req(struct as_req_state *state, krb5_error_code errcode)
      * by request keytype list.
      */
     client_key = NULL;
+    useenctype = 0;
     for (i = 0; i < state->request->nktypes; i++) {
         useenctype = state->request->ktype[i];
         if (!krb5_c_valid_enctype(useenctype))
diff --git a/src/lib/kadm5/srv/svr_iters.c b/src/lib/kadm5/srv/svr_iters.c
index b54dee3..d5a99de 100644
--- a/src/lib/kadm5/srv/svr_iters.c
+++ b/src/lib/kadm5/srv/svr_iters.c
@@ -187,7 +187,7 @@ static kadm5_ret_t kadm5_get_either(int princ,
 #ifdef BSD_REGEXPS
     char *msg;
 #endif
-    char *regexp;
+    char *regexp = NULL;
     int i, ret;
     kadm5_server_handle_t handle = server_handle;
 
diff --git a/src/slave/kprop.c b/src/slave/kprop.c
index 6fbfc1e..9f1c76a 100644
--- a/src/slave/kprop.c
+++ b/src/slave/kprop.c
@@ -310,7 +310,7 @@ void get_tickets(context)
 }
 
 static void
-open_connection(krb5_context context, char *host, int *fd)
+open_connection(krb5_context context, char *host, int *fd_out)
 {
     int     s;
     krb5_error_code retval;
@@ -320,6 +320,7 @@ open_connection(krb5_context context, char *host, int *fd)
     struct sockaddr_storage my_sin;
     int error;
 
+    *fd_out = -1;
     memset(&hints, 0, sizeof(hints));
     hints.ai_flags = AI_ADDRCONFIG;
     hints.ai_family = PF_UNSPEC;
@@ -348,7 +349,7 @@ open_connection(krb5_context context, char *host, int *fd)
         }
 
         /* We successfully connect()ed */
-        *fd = s;
+        *fd_out = s;
         retval = sockaddr2krbaddr(context, res->ai_family, res->ai_addr,
                                   &receiver_addr);
         if (retval != 0) {
